### DISCUSSION TITLE
Is the convert cli command slow?

### DISCUSSION THREAD

**Discussion by ktobah** [explosion/spaCy#8904] [Help: Coding & Implementations] [answered]
Hi,

I have a .conll file with over 3 million lines. I am trying to use the convert CLI command to convert my data into a .spacy file.

It has been running for over 20 hours. I would like to know if this is normal?

This is what I see on the terminal:

```
C:\spac>python -m spacy convert train_data.conll converted_data -c ner -n 10
ℹ Auto-detected token-per-line NER format
⚠ Document delimiters found, automatic document segmentation with `-n` disabled.

```

I am using windows 11, 32GB RAM, i7 CPU.

**Comment by polm** [comment]
Three million lines is a lot of data, but 20 hours does sound like too long. We had another issue about this that was solved by passing the `-s` option (#8490), but since you don't have a warning about it it seems like you already have sentence boundaries. 

In general conversion should be very fast, but there are some edge cases where it slows down a lot. I don't think we test conversion with any files that big.

I see there's a warning about your training file having document delimiters. Do you have really long documents? That could be a source of the problem, in which case you could fix it by just splitting up your documents.

**Comment by adrianeboyd** [comment, marked as answer]
In general, the CLI converters have not been optimized particularly well to deal with large inputs, although I wouldn't have expected 3 million lines to be a problem with 32 GB RAM. I tried an example with 30M lines (~30 MB on disk), which I could convert in ~1 minute using ~2 GB RAM on my laptop, so I'm not sure what might be going on. It's also possible certain combinations of sentence or document segmentation options lead to worse performance.

You can try splitting your .conll file up into a few shorter files to see if you are able to run the conversion successfully for those. If it works and you end up with multiple training files, you can provide a directory to `spacy train --paths.train train/` instead of a file and it will recursively search for files that end in `.spacy` in that directory.

**Comment by ktobah** [comment]
Hi, 

Thank you, guys.

@polm No, my documents are not that long, mostly 2 lines max.

@adrianeboyd I followed your suggestion and splitter the file into smaller chunks (100000 lines each). It is converting now, but still slow. It takes around 3-4 mins per file.

So, I am not sure if it is my system or something else. But I can work with 4 mins.
